sklearn/utils/_indexing.py

```diff
@@ -520,10 +532,18 @@ def resample(*arrays, replace=True, n_samples=None, random_state=None, stratify=
         )
 
     check_consistent_length(*arrays)
-
     if stratify is None:
         if replace:
-            indices = random_state.randint(0, n_samples, size=(max_n_samples,))
+            if sample_weight is not None:
+
+                p = sample_weight / sample_weight.sum()
+                indices = random_state.choice(
+                    np.arange(n_samples), size=(max_n_samples,), p=p
+                )
```

Comment:
Actually I think we can simplify it further:
```py
if sample_weight is not None:
    sample_weight = _check_sample_weight(sample_weight)
    p = sample_weight / sample_weight.sum()
else:
    p = None

indices = random_state.choice(n_samples, size=max_n_samples, p=p)
```
I even wonder if we should set explicitly replace=True even if it's the default for better readability